tools/test-js-with-node

```diff
@@ -291,12 +290,66 @@ def run_tests_via_node_js() -> int:
     ret = 0
     if parallel > 1:
         sub_tests = [test_files[i::parallel] for i in range(parallel)]
-        parallel_processes = [subprocess.Popen(node_tests_cmd + sub_test) for sub_test in sub_tests]
+        if options.coverage:
+            # ensure a clean start.
+            coverage_dir = os.path.join(ROOT_DIR, "var/node-coverage")
+            if os.path.exists(coverage_dir) and os.path.isdir(coverage_dir):
+                shutil.rmtree(coverage_dir)
+            # ensure that we work with a unique output folder just in case.
+            random_suffix = random.randrange(2 ** 63)
+            temp_dir = os.path.join(coverage_dir, f"nyc_outputs_{random_suffix}")
+            nyc = os.path.join(ROOT_DIR, "node_modules/.bin/nyc")
+            command = [nyc, "--extension", ".hbs", "--extension", ".ts"]
+            command += ["--silent"]
+            parallel_processes = []
+            for index, sub_test in enumerate(sub_tests):
+                command_with_unique_temp_dirs = command + [
+                    "--temp-directory",
+                    f"{temp_dir}/{index}",
+                ]
+                command_with_unique_temp_dirs += node_tests_cmd
+                parallel_processes += [subprocess.Popen(command_with_unique_temp_dirs + sub_test)]
```

Comment:
Nit: all the mutation makes it unnecessarily hard to follow this logic. Could just use a list comprehension.